tests/memory_tests.cpp

```diff
@@ -0,0 +1,202 @@
+#include "CppUTest/TestHarness.h"
+#include "CppUTestExt/MockSupport.h"
+#include "memory/general_memory_allocator.h"
+#include "memory/memory_region.h"
+#include "util/functions.h"
+#include <iostream>
+#include <stdlib.h>
+#define NUM_TEST_ALLOCATIONS 512
+#define MEM_SIZE 10000000
+uint32_t vtableAddress; // will hold address of the stealable test vtable
+
+class StealableTest : public Stealable {
+public:
+	void steal(char const* errorCode) { mock().actualCall("steal"); }
+	bool mayBeStolen(void* thingNotToStealFrom) { return true; }
+	int32_t getAppropriateQueue() { return 0; }
+	int32_t testIndex;
+};
+
+bool testReadingMemory(void* address, uint32_t size) {
```

Comment:
these utilities should be `static` functions to minimize the risk of mystery ODR violations in the future I think.

Reply:
Or if you want to be C++, put them in an anonymous namespace.